### DISCUSSION TITLE
maxHttpBufferSize and number of clients?

### DISCUSSION THREAD

**Discussion by neobie** [socketio/socket.io#5037] [General]
Is maxHttpBufferSize actually related to number of clients ?
EG, if I have 1000 clients connected, sending 1 message of 1KB at 1 time.
that would easily exceed 1MB as per defined in maxHttpBufferSize default value.

https://stackoverflow.com/a/12978658/309740
Here, it stated that the buffer is actually related to the message to be sent to the number of clients.

But in socket io documentation, https://socket.io/docs/v4/server-options/ 
it stated that "how many bytes a single message can be".

That could be confusing. So who is right ?

**Comment by darrachequesne** [comment]
> Is maxHttpBufferSize actually related to number of clients ?

No, it's not. The `maxHttpBufferSize` value is related to the size of a single message.

**Reply by neobie** [reply]
Thank you. However, I keep encountering an issue where Socket.IO drops the client's connection when suddenly a large number of messages need to be emitted in a very short time (e.g., 3000 messages per seconds to be emitted to clients based on the rooms they have joined). What could be causing this problem?

The CPU and Memory is not a problem, I have tried upscale to a very adequate resources with plenty of unused CPU and Memory, but the issue still happens.

Could there be a max buffer size a script can hold before it drops all the clients ? I have increased the NODE max heap size too and memory is not an issue.

**Reply by darrachequesne** [reply]
That's weird indeed. What is the reason of the disconnection? `transport close`? Are you able to reproduce the issue locally? Maybe some misconfigured proxy?

**Reply by neobie** [reply]
> That's weird indeed. What is the reason of the disconnection? `transport close`? Are you able to reproduce the issue locally? Maybe some misconfigured proxy?

It only happen when the incoming message is huge. Other time, when the messages are less than 1k per second, it is working fine. Not using proxy. 
Reason of disconnection ? Do you mean the error from client side?

**Reply by neobie** [reply]
> That's weird indeed. What is the reason of the disconnection? `transport close`? Are you able to reproduce the issue locally? Maybe some misconfigured proxy?

I get ping timeout or error timeout. 

I can confirm server resource itself is not the problem. I am running the same script in the same server with different port and serving only small amount of clients (less than 100), it is running well, sending messages on time, and doesn't drop the clients.
I can SSH to the server without any connection issue during that time.

Suspect some ubuntu param or nodejs socket io param that need to be tweaked to handle more messages per client, but I am running out of idea.

**Reply by llaakso** [reply]
Probably Socket.io is busy transferring your app data, so pings do not get through, timeout as a result. Test with longer timeout or/and phase sending data with small pause between.

**Reply by neobie** [reply]
Is there any logs / warning / error logs that tell you the clients has been dropped by socket io ?
That would be very useful to diagnosis the if the servers need more resources.
